src/components/spreadsheet/spreadsheet-gs-filter.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Badge, Button, Popover } from '@mui/material';
+import ArticleIcon from '@mui/icons-material/Article';
+import { FormattedMessage } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { CustomFormProvider, DirectoryItemsInput, ElementType, useStateBoolean } from '@gridsuite/commons-ui';
+import { AppState } from '../../redux/reducer';
+import { saveSpreadsheetGsFilters } from '../../redux/actions';
+import { ExpertFilter } from '../../services/study/filter';
+import { SpreadsheetEquipmentType } from './config/spreadsheet.type';
+import {
+    toExpertFilter,
+    toFormFormat,
+    initialSpreadsheetGsFilterForm,
+    SpreadsheetGsFilterForm,
+    spreadsheetGsFilterFormSchema,
+} from './utils/spreadsheet-gs-filter-utils';
+import { SPREADSHEET_GS_FILTER } from '../utils/field-constants';
+import { spreadsheetStyles } from './utils/style';
+
+interface SpreadsheetGsFilterProps {
+    equipmentType: SpreadsheetEquipmentType;
+    applyGsFilter: (filter: ExpertFilter[]) => void;
+}
+
+export const SpreadsheetGsFilter = ({ equipmentType, applyGsFilter }: SpreadsheetGsFilterProps) => {
+    const dispatch = useDispatch();
+    const gsFilterSpreadsheetState = useSelector((state: AppState) => state.gsFilterSpreadsheetState);
```

Comment:
Maybe also move this in the hook ?

Reply:
I moved the filter application logic inside the hook but had to keep the reference here to be able to reset the input values when switching equipment